file-formats/aifp/type/zif_aff_aifp_v1.intf.abap

```diff
@@ -0,0 +1,223 @@
+      "! <p class="shorttext">Message Variable 2</p>
+      "! Message variable 2
+      message_variable_2 TYPE string,
+      "! <p class="shorttext">Message Variable 3</p>
+      "! Message variable 3
+      message_variable_3 TYPE string,
+      "! <p class="shorttext">Message Variable 4</p>
+      "! Message variable 4
+      message_variable_4 TYPE string,
+    END OF ty_message.
+  TYPES:
+    "! <p class="shorttext">Field Check</p>
+    "! Field check
+    BEGIN OF ty_field_check,
+      "! <p class="shorttext">Field Check</p>
+      "! Field check
+      field_check TYPE ty_simple_field_check,
+      "! <p class="shorttext">Operator</p>
+      "! Operator for field check
+      operator    TYPE zif_aff_types_v1=>ty_option,
+      "! <p class="shorttext">Pattern</p>
+      "! Pattern for Field Check
+      pattern     TYPE c LENGTH 60,
+    END OF ty_field_check.
+  TYPES:
+    "! <p class="shorttext">Database Check</p>
+    "! Database check
+    BEGIN OF ty_database_check,
+      "! <p class="shorttext">Table Name</p>
+      "! Table name
+      table_name      TYPE c LENGTH 45,
+      "! <p class="shorttext">Where Condition for Select Statement</p>
+      "! Where Condition for Select Statement
+      where_condition TYPE c LENGTH 200,
+      "! <p class="shorttext">Check Type</p>
+      "! Check type
+      check_type      TYPE ty_database_check_type,
+      "! <p class="shorttext">Field Name for Comparison</p>
+      "! Field Name for Comparison
+      field_name      TYPE c LENGTH 45,
+      "! <p class="shorttext">Operator</p>
+      "! Operator for database check
+      operator        TYPE zif_aff_types_v1=>ty_option,
+      "! <p class="shorttext">Pattern</p>
+      "! Pattern for database check
+      pattern         TYPE c LENGTH 60,
+    END OF ty_database_check.
+  TYPES:
+    "! <p class="shorttext">Class Check</p>
+    "! Class check
+    BEGIN OF ty_class_check,
+      "! <p class="shorttext">Class</p>
+      "! Class name for check
+      class TYPE zif_aff_types_v1=>ty_object_name_30,
+    END OF ty_class_check.
+  TYPES:
+    "! <p class="shorttext">Single Check</p>
+    "! Single check
+    BEGIN OF ty_single_check,
+      "! <p class="shorttext">Number</p>
+      "! Number
+      "! $required
+      number          TYPE n LENGTH 3,
+      "! <p class="shorttext">Description</p>
+      "! Description
+      description     TYPE c LENGTH 120,
+      "! <p class="shorttext">Scenario</p>
+      "! Scenario
+      scenario        TYPE ty_check_scenario,
+      "! <p class="shorttext">Field Check</p>
+      "! Field check
+      field_check     TYPE ty_field_check,
+      "! <p class="shorttext">Database Check</p>
+      "! Database check
+      database_check  TYPE ty_database_check,
+      "! <p class="shorttext">Class Check</p>
+      "! Class check
+      class_check     TYPE ty_class_check,
+      "! <p class="shorttext">Success Message</p>
+      "! Success message
+      success_message TYPE ty_message,
```

Comment:
why is the success_message on another level as the error_message?

Reply:
@huber-nicolas, a check consists of several individual checks, so called single checks, connected by an OR operator. It is possible to assign a success message to each single check, so that it can be determined which check was successful.
If none of the single checks is successful, a overall error message is thrown, which is defined at the check level.